src/main/java/com/example/onlinebookstore/repository/BookRepository.java

```diff
@@ -1,13 +1,9 @@
 package com.example.onlinebookstore.repository;
 
 import com.example.onlinebookstore.model.Book;
-import java.util.List;
 import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> findById(Long id);
+public interface BookRepository extends JpaRepository<Book, Long> {
```

Comment:
Annotation Repository is missing

Reply:
do we really have to put it above the interface? but I will do it)

Reply:
As you extend JpaRepository it's not required, but it's good if we have this annotation.